src/sp/it/pl/audio/Item.kt

```diff
@@ -23,10 +25,9 @@ abstract class Item {
     fun isFileBased(): Boolean = "file"==uri.scheme
 
     /**
-     * @return absolute file this item represents
-     * @throws RuntimeException if this item is not file based
+     * @return absolute file this item represents or null if it is not based on file
      */
-    open fun getFile() = uri.toFileOrNull()!!.absoluteFile!!
+    open fun getFile(): File? = if (isFileBased()) uri.toFileOrNull()!!.absoluteFile else  null
```

Comment:
The proposed solution looks nice.

My solution however makes the potential failure explicit, which is a good thing. One thing I learned is that URI.create(), new URI(), new File(), etc are very confusing. These are fairly outdated and there is a bit of inconsistency in when and how they end up throwing. I actually deprecated new File(URI) for this reason. That is why the URI.toFileOrNull() exists - for cases where we do not care about the exception, because we expect it to work every single time. It is exactly the same as String.toIntOrNull methods provided by Kotlin stdlib.

Which is why I prefer my solution, but the consuming of the exception does make me little concerned. One can never be too sure. The file uri in item should always be well formed, but...

I'll think about this a bit.

Reply:
You could create a File.toUrlOrThrow or smth

Reply:
Will be solved in #85. There is more methods affected and thus out of scope of this PR.

Reply:
but Try swallows the exception as well?

Reply:
Try reports error, type of which depends on the design, in this case Try<URL,Exception>. But that will be solved in next PR, there is indeed some head-scratching around this.